homekit/ble_impl/device.py

```diff
@@ -0,0 +1,121 @@
+import collections
+import logging
+import time
+
+import dbus
+from . import gatt
+from .manufacturer_data import parse_manufacturer_specific
+
+
+logger = logging.getLogger(__name__)
+
+
+class Device(gatt.Device):
+
+    def __init__(self, *args, **kwargs):
+        gatt.Device.__init__(self, *args, **kwargs, managed=False)
+
+        self.subscribers = collections.defaultdict(list)
+        self.homekit_discovery_data = self.get_homekit_discovery_data()
+
+    def __repr__(self):
+        return 'BleDevice[mac_address="{}", name="{}"]'.format(self.mac_address, self.name)
+
+    @property
+    def name(self):
+        return self._properties.Get('org.bluez.Device1', 'Alias')
+
+    def get_homekit_discovery_data(self):
+        try:
+            mfr_data = self._properties.Get('org.bluez.Device1', 'ManufacturerData')
+        except dbus.exceptions.DBusException as e:
+            if e.get_dbus_name() == 'org.freedesktop.DBus.Error.InvalidArgs':
+                return {}
+            raise
+
+        mfr_data = dict((int(k), bytes(bytearray(v))) for (k, v) in mfr_data.items())
+
+        if 76 not in mfr_data:
+            return
+
+        parsed = parse_manufacturer_specific(mfr_data[76])
+
+        if parsed['type'] != 'HomeKit':
+            return
+
+        return parsed
+
+    def connect(self):
+        super().connect()
+
+        try:
+            if not self.services:
+                logger.debug('waiting for services to be resolved')
+                for i in range(20):
+                    if self.is_services_resolved():
+                        break
+                    time.sleep(1)
+                else:
+                    raise AccessoryNotFoundError('Unable to resolve device services + characteristics')
```

Comment:
add `from homekit.exceptions import AccessoryNotFoundError` to import the Exception